ApiBundle/Transformer/WidgetCollectionTransformer.php

```diff
@@ -34,4 +40,22 @@ public function getName()
     {
         return 'widget_collection';
     }
+
+    /**
+     * @param $widget
+     *
+     * @return null|string
+     */
+    protected function getWidgetRole($widget)
+    {
+        $role = null;
+
+        if (preg_match('/node/', $widget)) {
```

Comment:
it's not extensible and it's work only with widget of node and content. an integrator can be add a custom widget so your code not works with her widget